Control/Monad/Reader/Class.hs

```diff
@@ -202,3 +206,17 @@ instance
       r <- ask
       local f (runSelectT m (local (const r) . c))
     reader = lift . reader
+
+type LiftingReader :: ((Type -> Type) -> Type -> Type) -> (Type -> Type) -> Type -> Type
+newtype LiftingReader t m a = LiftingReader (t m a)
+  deriving (Functor, Applicative, Monad, MonadTrans)
+
+instance MonadReader r m => MonadReader r (LiftingReader (ReaderT r') m) where
+  ask = lift ask
+  local f (LiftingReader (ReaderT.ReaderT x)) = LiftingReader . ReaderT.ReaderT $ local f . x
+  reader = lift . reader
+
+instance (MonadReader r m, Monoid w) => MonadReader r (LiftingReader (LazyRWS.RWST r' w s) m) where
+  ask = lift ask
+  local f (LiftingReader (LazyRWS.RWST x)) = LiftingReader . LazyRWS.RWST $ \r s -> local f $ x r s
+  reader = lift . reader
```

Comment:
```suggestion
instance MonadReader r m => MonadReader r (LiftingReader (ReaderT r') m) where
  ask = lift ask
  local f = mapLiftingReader (ReaderT.mapReaderT (local f))
  reader = lift . reader

instance (MonadReader r m, Monoid w) => MonadReader r (LiftingReader (LazyRWS.RWST r' w s) m) where
  ask = lift ask
  local f = mapLiftingReader (LazyRWS.mapRWST (local f))
  reader = lift . reader

mapLiftingReader :: (t m a -> t n b) -> LiftingReader t m a -> LiftingReader t n b
mapLiftingReader = coerce
```

Reply:
This github diff view is terrible. Did you only change lines 216 and 221 with mapLiftingReader?

Reply:
Yup.